models/schemas/recordSchema.ts

```diff
@@ -27,3 +29,23 @@ const RecordSchema = new mongoose.Schema<Record>({
 });
 
 export default RecordSchema;
+
+// On save, call refreshIndexCalcs on the level with levelId
+RecordSchema.post('save', async function() {
+  const level = await LevelModel.findById(this.levelId);
+
+  if (level) {
+    await refreshIndexCalcs(level);
```

Comment:
instead of this, maybe you can just do something like:

`LevelModel.updateOne({ _id: this.levelId }, { $set: { calc_records_last_ts: this.ts}, $inc: { calc_records_count: 1 } })`

Reply:
i just get worried about a DB error or something happening and we miss a count. maybe if we had a job that checked for errors in the calcs every night?